fragmentshandler/fragmentshandler/src/androidTest/java/com/microsoft/device/dualscreen/fragmentshandler/utils/SampleActivity.kt

```diff
@@ -74,15 +73,14 @@ class SampleActivity : AppCompatActivity() {
     }
 
     private fun registerWindowInfoFlow() {
-        val windowInfoRepository = windowInfoRepository()
-        lifecycleScope.launch(Dispatchers.Main) {
-            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                windowInfoRepository.windowLayoutInfo.collectIndexed { index, info ->
+        job = MainScope().launch {
```

Comment:
Why did you changed to coroutines and job scheduler? With the lifecycle stuff, the callback it's added and removed automatically. Now, you need to cancel that job somewhere:
From my side, the correct version should be:
// Create a new coroutine since repeatOnLifecycle is a suspend function
        lifecycleScope.launch(Dispatchers.Main) {
            // The block passed to repeatOnLifecycle is executed when the lifecycle
            // is at least STARTED and is cancelled when the lifecycle is STOPPED.
            // It automatically restarts the block when the lifecycle is STARTED again.
            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
                // Safely collects from windowInfoRepository when the lifecycle is STARTED
                // and stops collection when the lifecycle is STOPPED.
                WindowInfoTracker.getOrCreate(this@DisplayFeaturesActivity)
                    .windowLayoutInfo(this@DisplayFeaturesActivity)
                    .collect { newLayoutInfo ->
                        ....
                    }
            }
        }

Reply:
Totally agree. Using lifecycle when possible, and in our case is since we were using it already, it's the recommended way to use WM (the benefits Andrei has mentioned about lifecycle automatic work + is what Google and us are showing on samples, blog posts, etc.).

Reply:
Fixed